### DISCUSSION TITLE
Allow changing Nova resource path

### DISCUSSION THREAD

**Discussion by th-alanreed** [laravel/nova-issues#5801] [Ideas & Feature Requests]
There should be a way to change the resource path from `/nova/resources/articles/3` to `/nova/resources/articles/my-article-slug`

**Comment by davidhemphill** [comment]
Interested in your use-case...could you go into more detail?

**Comment by th-alanreed** [comment]
There are a couple reasons:
1) Our nova application is client-facing so having a more contextual URL path is nice. This is the same reason you can customize the key when using Route-Model binding. 
2) It would be easier to link directly to nova resources from outside of the primary application

It ought to just work when you define `getRouteKeyName()` on the model. https://laravel.com/docs/10.x/routing#customizing-the-default-key-name

**Reply by davidhemphill** [reply]
Internally Nova uses `whereKey` to fetch the resource by its primary key, which guarantees uniqueness. 

Additionally, Nova tries to avoid requiring users to modify their Eloquent models when possible, which is why `getRouteKeyName` isn't picked up.

**Reply by th-alanreed** [reply]
Makes sense to avoid requiring changes to Eloquent models. That's really smart. 

There should still be a way to override `whereKey` then. It's the developer's responsibility to ensure the column is unique.

**Reply by crynobone** [reply]
If the idea and expected behaviours of resource route key is only limited to using alternative structure then it is possible to make a PR just for that. However, if the idea is to hide fhe ID (security etc) it now become a major revamps.

Some other limitations if we can want to add the former uss case:

1. Lens will still need to use ID since it doesn't always contain the alternative slug, existing application may be broken if we force this and. Lens to resource detail page will be broken as resource expect alternative route key while lens only can guarantee ID.
2. the changes is limited to URL while `nova-api` will still use ID to avoid major revamp.
3. ID still be used and refer to the primary key.
